src/libs/antares/constants.h

```diff
@@ -44,6 +44,11 @@
 /*! */
 /* # define ANTARES_USE_GLOBAL_MAXIMUM_COST 0 */
 
+// The maximum number of areas allowed in a study
+#define MAX_NUMBER_OF_AREAS 9999
+// The maximum number of thermal plants per area allowed in a study
+#define MAX_NUMBER_OF_THERMAL_CLUSTERS_PER_AREA 999
```

Comment:
It is my opinion that there should be no arbitrary limit on the number of areas or clusters per area.

While I agree that human-made studies will almost never go beyond these limits, computer-generated studies could.